app-typescript/components/TreeSelect/TreeSelect.tsx

```diff
@@ -96,6 +96,12 @@ export class TreeSelect<T> extends React.Component<IProps<T>, IState<T>> {
     private popperInstance: Instance | null;
     private zIndex: number = getNextZIndex();
 
+    /*
+    * This variable is used to distinguish changes coming from outside (from props.value)
+    * from those made by the user through interaction with the component.
+    */
```

Comment:
Include in the comment is that the purpose is to prevent triggering onChange when props change.